.github/workflows/ci.yml

```diff
@@ -11,30 +11,14 @@ on:
     types: [opened, synchronize]
 
 jobs:
-  main:
+  ci:
     runs-on: ubuntu-latest
-    permissions:
-      packages: write
-      contents: read
     steps:
-      - uses: actions/checkout@v1
-      - name: Set up JDK 1.8
-        uses: actions/setup-java@v1
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
         with:
```

Comment:
https://github.com/actions/setup-java/blob/main/docs/switching-to-v2.md#dropping-legacy-java-version-syntax-1x
```
V1 supported legacy Java syntax such as 1.8 (same as 8) and 1.8.0.212 (same as 8.0.212).
V2 dropped support for legacy syntax so workflows should be updated accordingly.
```